### ISSUE TITLE
[api-extractor] Internal getter for internal type produces error if setter also defined

### ISSUE THREAD

**Issue by pmconne** [microsoft/rushstack#1679]
**Is this a feature or a bug?**

- [ ] Feature
- [X] Bug

**Please describe the actual behavior.**

See repro steps below. Regression introduced in #1646 

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

```ts
/** @internal */
export class Internal {
}

/** @public */
export class Public {
  private _internal: Internal = { };

  /** @internal */
  public get internal(): Internal { return this._internal; }
  public set internal(internal: Internal) { this._internal = internal; }
}
```

Produces:
`(ae-incompatible-release-tags) The symbol "internal" is marked as @public, but its signature references "Internal" which is marked as @internal`

But adding `@internal` to the setter produces (as expected):
`(ae-setter-with-docs) The doc comment for the property "internal" must appear on the getter, not the setter.`

**What is the expected behavior?**

No error should be produced. The release tag, like the documentation, should be defined solely by the getter. (Though IMO it should be permitted to have different release tag on setter vs getter - that's a separate issue, but relevant to this one).

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** api-extractor
* **Tool Version:** 7.7.0
* **Node Version:** 10.15.0
  * **Is this a LTS version?** Yes 
  * **Have you tested on a LTS version?** 
* **OS:** Windows 10

**Comment by octogonz** [comment]
I was able to repro this.  It's definitely a bug.  

The setter is modeled as an "ancillary" declaration, and the getter is the "main" declaration.

Normally the analyzer simply ignores ancillary declarations.  However, `ae-incompatible-release-tags` really does need to to analyze both declarations.  But in the case of the setter, it should be using the release tag from the getter (whereas currently the setters release tag is calculated independently).

This is not hard to fix, but to prevent future mistakes like this, we should probably do a little refactoring.  It seems that `DeclarationMetadata.isAncillary` probably shouldn't have been part of `DeclarationMetadata`, since it determines where your metadata comes from.  Here's some possible alternatives for how that could be implemented:

1. Ancillary declarations should not store their own `DeclarationMetadata` fields, and callers should look to the main declaration for this info; OR
2. Ancillary declarations should share their `DeclarationMetadata` object with the main declaration; OR
3. Ancillary declarations should copy the relevant metadata fields from the main declaration (but keep this in sync somehow)

I'm leaning towards 1, but let me think about it a bit.

@pmconne Thank you for providing a repro, super helpful!

**Comment by octogonz** [comment]
Here's a PR: https://github.com/microsoft/rushstack/pull/1681

I ended up going with a hybrid approach, separating the non-ancillary portion of `DeclarationMetadata` into a new class `SignatureMetadata`.

**Comment by octogonz** [comment]
@pmconne This fix was published with API Extractor 7.7.3; please upgrade and let us know if your problem is solved. Thanks!
